syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/any.proto";
import "google/protobuf/wrappers.proto";

package doodle;

message fileInfo{
  string path = 1;
  bool idFolder = 2;
  bool exist = 3;
  google.protobuf.Timestamp update_time = 4;
  uint64 size = 5;
}

message fileStream{
  fileInfo info = 1;
  string path = 2;
  uint64 beg = 3;
  uint64 end = 4;
  google.protobuf.Any data = 5;

}

service FileSystemServer{
  rpc GetInfo(fileInfo) returns (fileInfo) {}
  
  rpc GetSize(fileInfo) returns (fileInfo) {}
  rpc IsFolder(fileInfo) returns (fileInfo) {}
  rpc GetTimestamp(fileInfo) returns (fileInfo) {}
  rpc IsExist(fileInfo) returns (fileInfo) {}

  rpc Download(fileInfo) returns (stream fileStream) {}
  rpc Upload(stream fileStream) returns (fileInfo) {}
}
